autofs-5.1.8 - add missing include to log.h for pid_t

From: Sam James <sam@gentoo.org>

Fixes build failures on musl like:
```
../include/log.h:49:8: error: unknown type name 'pid_t'
   49 | extern pid_t log_pidinfo(struct autofs_point *ap, pid_t pid, char *label);
      |        ^~~~~
../include/log.h:49:51: error: unknown type name 'pid_t'; did you mean 'gid_t'?
   49 | extern pid_t log_pidinfo(struct autofs_point *ap, pid_t pid, char *label);
      |                                                   ^~~~~
      |                                                   gid_t
```

Tested-by: Yixun Lan <dlan@gentoo.org>
Signed-off-by: Sam James <sam@gentoo.org>
---
 CHANGELOG     |    1 +
 include/log.h |    2 ++
 2 files changed, 3 insertions(+)

diff --git a/CHANGELOG b/CHANGELOG
index 02f8ef5f..794cd04b 100644
--- a/CHANGELOG
+++ b/CHANGELOG
@@ -22,6 +22,7 @@
 - musl: define fallback dummy NSS config path
 - musl: avoid internal stat.h definitions.
 - musl: add missing include to hash.h for _WORDSIZE.
+- musl: add missing include to log.h for pid_t.
 
 19/10/2021 autofs-5.1.8
 - add xdr_exports().
diff --git a/include/log.h b/include/log.h
index 69eed96b..a7b09f92 100644
--- a/include/log.h
+++ b/include/log.h
@@ -17,6 +17,8 @@
 #ifndef LOG_H
 #define LOG_H
 
+#include <unistd.h>
+
 /* Define logging functions */
 
 #define LOGOPT_NONE	0x0000
